src/parser/mod.rs

```diff
@@ -5894,6 +5896,47 @@ impl<'a> Parser<'a> {
         Ok(owner)
     }
 
+    /// ```sql
+    ///     CREATE DOMAIN name [ AS ] data_type
+    ///             [ COLLATE collation ]
+    ///             [ DEFAULT expression ]
+    ///             [ domain_constraint [ ... ] ]
+    ///     
+    ///         where domain_constraint is:
+    ///     
+    ///         [ CONSTRAINT constraint_name ]
+    ///         { NOT NULL | NULL | CHECK (expression) }
+    /// ```
+    ///
+    /// [PostgreSQL Documentation](https://www.postgresql.org/docs/current/sql-createdomain.html)
```

Comment:
same here regarding duplicating the doc comment, we can only link to `[Statement::CreateDomain]` which readers can follow for more context around the statement if needed.

Reply:
Done